src/react/DeviceStatusPage/DeviceLiveEventsCard.tsx

```diff
@@ -0,0 +1,347 @@
+        </Badge>
+        <span>Device disconnected</span>
+      </>
+    );
+  }
+  if (event instanceof AstarteDeviceIncomingDataEvent) {
+    return (
+      <>
+        <Badge variant="info" className="mr-2">
+          incoming data
+        </Badge>
+        <span className="mr-2">{event.interfaceName}</span>
+        <span className="mr-2">{event.path}</span>
+        <span className="mr-2 text-monospace">{event.value}</span>
+      </>
+    )
+  }
+  return <></>;
+};
+
+interface AstarteDeviceEventDelegateProps {
+  event: AstarteDeviceEvent;
+}
+
+const AstarteDeviceEventDelegate = ({ event }: AstarteDeviceEventDelegateProps) => {
+  return (
+    <li className="event-device px-2">
+      <Timestamp>{new Date(event.timestamp)}</Timestamp>
+      { astarteDeviceEventBody(event) }
+    </li>
+  );
+};
+
+interface SystemEventDelegateProps {
+  event: SystemEvent;
+}
+
+const SystemEventDelegate = ({ event }: SystemEventDelegateProps) => {
+  switch (event.level) {
+    case 'error':
+      return (
+        <li className="px-2">
+          <Timestamp>{new Date(event.timestamp)}</Timestamp>
+          <Badge variant="secondary" className="mr-2">
+            channel
+          </Badge>
+          <span className="text-danger">{event.message}</span>
+        </li>
+      );
+
+    case 'info':
+    default:
+      return (
+        <li className="px-2">
+          <Timestamp>{new Date(event.timestamp)}</Timestamp>
+          <Badge variant="secondary" className="mr-2">
+            channel
+          </Badge>
+          <span className="text-secondary">{event.message}</span>
+        </li>
+      );
+  }
+};
+
+interface EventDelegateProps {
+  event: RenderableEvent;
+}
+
+const EventDelegate = ({ event }: EventDelegateProps) => {
+  if (isSystemEvent(event)) {
+    return <SystemEventDelegate event={event} />;
+  }
+
+  return <AstarteDeviceEventDelegate event={event} />;
+};
+
+interface DeviceLiveEventsViewProps {
+  astarte: AstarteClient;
+  deviceId: string;
+}
+
+const DeviceLiveEventsView = ({
+  astarte,
+  deviceId,
+}: DeviceLiveEventsViewProps): React.ReactElement => {
+  const [deviceEvents, setDeviceEvents] = useState<RenderableEvent[]>([]);
+
+  const registerEvent = (event: RenderableEvent) => {
+    setDeviceEvents((oldEvents: RenderableEvent[]) => {
+      const newEvents = oldEvents.slice();
+      newEvents.push(event);
+      return newEvents;
+    });
+  };
+
+  const sendErrorMessage = (errorMessage: string) =>
+    registerEvent({
+      message: errorMessage,
+      level: 'error',
+      timestamp: Date.now(),
+    });
+
+  const sendInfoMessage = (infoMessage: string) =>
+    registerEvent({
+      message: infoMessage,
+      level: 'info',
+      timestamp: Date.now(),
+    });
+
+  useEffect(() => {
+    const onSocketErrorCB = () => sendErrorMessage('Astarte channels communication error');
+    const onSocketCloseCB = () => sendErrorMessage('Lost connection with the Astarte channel');
```

Comment:
I would stay away from odd abbreviations, just `onSocketError` / `onSocketClose` would be fine I guess